kernels/ZendEngine2/fcall.c

```diff
@@ -564,12 +564,13 @@ int zephir_call_user_function(zval **object_pp, zend_class_entry *obj_ce, zephir
 	//	fcic.function_handler->op_array.line_end = 0;
 	//}
 
-	/* fcic.initialized = 0; */
-#if PHP_VERSION_ID >= 50600
-	status = ZEPHIR_ZEND_CALL_FUNCTION_WRAPPER(&fci, &fcic, info TSRMLS_CC);
-#else
-	status = ZEPHIR_ZEND_CALL_FUNCTION_WRAPPER(&fci, &fcic TSRMLS_CC);
-#endif
+	fcic.initialized = 0;
+	status = zend_call_function(&fci, &fcic TSRMLS_CC);
```

Comment:
@sjinks Just inspected this in ZE2 and ZE3.

**ZE2** (before `zend_call_function`):
```
print fcic.called_scope.name.val   # (zend_class_entry *) 0x0
print fcic.calling_scope.name.val  # (zend_class_entry *) 0x0
print fcic.object_ptr              # (zval *) 0x0
print print fcic.function_handler  # (zend_function *) 0x0
```

**ZE3** (before `zend_call_function`):
```
printf "%s\n", fcic.called_scope.name.val                     # Group\PrivateCaller
printf "%s\n", fcic.calling_scope.name.val                    # Group\PrivateCaller
printf "%s\n", fcic.object.ce.name.val                        # Group\PrivateCaller
printf "%s\n", fcic.function_handler.common.function_name.val # runBar
```

Actually I like your solution with `populate_fcic` and `resolve_callable` for ZE3. Could you please try to backport they to the ZE2?